finance_dl/amazon.py

```diff
@@ -122,8 +133,13 @@ def get_invoice_urls():
             while True:
 
                 def invoice_finder():
-                    return self.find_elements_by_descendant_text_match(
-                        '. = "Invoice"', 'a', only_displayed=True)
+                    if self.amazon_domain == Domain.CO_UK:
+                        # on amazon.co.uk there are no direct invoice links
+                        # clicking causes some js to execute and show popup window
+                        return self.driver.find_elements(By.XPATH, '//a[contains(@href, "summary/print.html")]')
```

Comment:
It looks like this might also work for both domains, right?  In which case we could simplify the code.

Reply:
I guess I wanted to keep the original code as intact as possible, e.g. in case they change the link to something else. In the sense finding an 'Invoice' element on the screen and clicking is more stable. But up to you, I can change both to looks for 'summary/print.html'.

Reply:
That makes sense (although it does depend on the site), but I think it is very useful to keep the code paths as converged as possible to make testing more feasible.  It looks like the way amazon works is that you can log into any site with the same credentials but the orders are independent, so in practice one person will only be able to test a single site.  Better that we are testing the same code as much as possible.